docs/developer/plugins.md

```diff
@@ -151,6 +151,38 @@ process foo {
 Refer to the source code of Nextflow's built-in executors to see how to implement the various components of an executor. You might be able to implement most of your executor by simply reusing existing code.
 :::
 
+### Filesystems
+
+Plugins can define custom filesystems, which are used by Nextflow to interact with external storage systems through a single interface (see {ref}`remote-files`).
+
+To implement a custom filesystem, create a class in your plugin that extends [`FileSystemProvider`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/nio/file/spi/FileSystemProvider.html). Implement the `getScheme()` method to define the URI scheme for your filesystem:
+
+```groovy
+import java.nio.file.spi.FileSystemProvider
+
+class MyFileSystemProvider extends FileSystemProvider {
+
+    @Override
+    String getScheme() {
+        return 'myfs'
+    }
+
+    // ...
+}
+```
```

Comment:
I'd just mention Nextflow relies in on Java NIO 2 API and link some example e.g. http file system. The implementation is out of the scope of Nextflow documentation. 

The above example it's far from being useful.

Reply:
The example is meant to show the connection between defining the URI scheme in the implementation and then using it in Nextflow. Similar to the Executor example above

Reply:
Fair enough, then this link can provide more context https://docs.oracle.com/javase/8/docs/technotes/guides/io/fsp/filesystemprovider.html